### PR TITLE
Redirect user to unauthorized page

### LINKED ISSUE #69: redirect user to unauthorized page if they attempt to access restricted material
**As a** authenticated user
**I want** to be forwarded to an "unauthorized page" when attempting to access a restricted page
**So that** I know that where I tried to go is not accessible

**Acceptance Criteria**
If a user doesn't have the privileges to see a page the attempt to navigate to, send them to an unauthorized page that tells them to contact the site administrator to gain access.  
(An example would be if a non-admin user attempted to navigate to the controlled vocabulary page #38 ).

### PR DESCRIPTION
**Pull request by AScaduto**
Closes #69 

Unauthorized page added to router; All unauthenticated users will now redirect to /unauthorized when attempting to reach a permissions restricted page

Can be tested via corresponding e2e test or by hitting a restricted page i.e /substances /vocabularies with a test user with the django permission, is_superuser set to false

### CODE REVIEW

**Review by michael-barbour — COMMENTED** [review]
> An example would be if a non-admin user attempted to navigate to the controlled vocabulary page #38

The page visually looks good but it seems as if this requires the functionality of checking to see whether the user is a super user.

File: src/router/index.js [hunk 1]
```diff
@@ -47,7 +55,7 @@ router.beforeEach(async (to, from, next) => {
     await store.dispatch("auth/fetchUser");
     if (!store.getters["auth/isAuthenticated"]) {
       next({
-        name: "home"
+        name: "unauthorized"
```
Comment (michael-barbour) [inline src/router/index.js:58]: Authentication and Authorization are different.  When your authentication is expired the user should be prompted to sign in again.

Unless I'm misunderstanding, a user should only be directed to the authorization page when accessing a resource that the user doesn't have permission to see.

File: src/router/index.js [hunk 2]
```diff
@@ -31,7 +31,15 @@ const routes = [
   {
     path: "/lists",
     name: "lists",
-    component: () => import("../views/Lists")
+    component: () => import("../views/Lists"),
+    meta: {
+      requiresAuth: true
+    }
+  },
```
Comment (michael-barbour) [inline src/router/index.js:38]: > As a anonymous user

#51 The list viewing page should be available for unauthenticated users.

**Review by michael-barbour — CHANGES_REQUESTED** [review]
This requires a check on is_superuser from the user resource.  If it fails any restricted page should redirect (currently this is only the /vocabularies/ page)

File: src/router/index.js (hunk 2 above)
Comment (AScaduto) [reply src/router/index.js]: :+1:

File: src/router/index.js (hunk 1 above)
Comment (AScaduto) [reply src/router/index.js:60]: Made revisions; Non logged in users will be sent back to the homepage but those that aren't superusers will be able to select other pages from the navbar but those that they do not have the appropriate permissions to access will show up as the 'unauthorized' page but not log them out or send them back to the homepage locking them out of the site without having to press the back button.

image

Comment (AScaduto) [reply src/router/index.js:60]: Second revision: 

Unauthorized pages will no longer show in navbar 

image

**Review by michael-barbour — APPROVED** [review]
These changes look good.  Currently the substance page is locked behind `requiresAuth` despite the ticket #54 saying this should be available to anonymous users.

**Pull request merged**